students/DennisLee/lesson06/test_mailroom.py

```diff
@@ -0,0 +1,393 @@
+            ) == True
+
+def test_call_menu_function_103():
+    # An I/O error occurs because the function prompts for donation amt
+    with pytest.raises(IOError):
+        m.call_menu_function(alt_choices, 'C. N. Emone', m.get_donation_amount)
+
+def test_call_menu_function_104():
+    # An I/O error occurs because the function prompts for donation amt
+    with pytest.raises(IOError):
+        m.call_menu_function(alt_choices, 'Papa Smurf', m.get_donation_amount)
+
+
+def test_get_donation_amount():
+    # An I/O error occurs because the function prompts for a donation
+    # amount or to quit
+    with pytest.raises(IOError):
+        m.get_donation_amount('C. N. Enome')
+    
+# Simulate and test function calls that occur in the code 
+# after the input() function that causes the I/O error
+# '' and 'quit' are in the choice dict, so they return True
+# All other simulated input for the `call_menu_function` call returns
+# False from that function, regardless of whether the `add_donation`
+# call successfully adds a donation amount to the donor history list
+def test_call_menu_function_200():
+    # re-use the donor name dict, except without the 'list' function
+    donation_choices = alt_choices.copy()
+    donation_choices.pop('list')
+    assert m.call_menu_function(donation_choices, '', m.add_donation,
+            donor_name='C. N. Enome') == True
+    
+def test_call_menu_function_201():
+    donation_choices = alt_choices.copy()
+    donation_choices.pop('list')
+    assert m.call_menu_function(donation_choices, 'quit', m.add_donation,
+            donor_name='C. N. Enome') == True
+    
+def test_call_menu_function_202():
+    donation_choices = alt_choices.copy()
+    donation_choices.pop('list')
+    assert m.call_menu_function(donation_choices, '50', m.add_donation,
+            donor_name='C. N. Enome') == False
+    del m.donor_history['C. N. Enome']  # restore dict for next tests
+
+def test_call_menu_function_203():
+    donation_choices = alt_choices.copy()
+    donation_choices.pop('list')
+    assert m.call_menu_function(donation_choices, '0.00', m.add_donation,
+            donor_name='C. N. Enome') == False
+    
+def test_call_menu_function_204():
+    donation_choices = alt_choices.copy()
+    donation_choices.pop('list')
+    assert m.call_menu_function(donation_choices, '-100.0045', m.add_donation,
+            donor_name='C. N. Enome') == False
+    
+def test_call_menu_function_205():
+    donation_choices = alt_choices.copy()
+    donation_choices.pop('list')
+    assert m.call_menu_function(donation_choices, 'fdasfd', m.add_donation,
+            donor_name='C. N. Enome') == False
+
+
+# For the `add_donation` call, only positive numbers are added
+# successfully to the donor history list and are returned back
+def test_add_donation_1():
+    assert m.add_donation(50, 'C. N. Enome') == 50
+    del m.donor_history['C. N. Enome']  # restore dict for next tests
```

Comment:
The `unittest` framework includes a `setUp()` method and some other helper methods for shared test data to help you out with these kinds of situations